builder/dockerfile/internals.go

```diff
@@ -569,8 +575,13 @@ func (b *Builder) run(cID string) (err error) {
 			Code:    ret,
 		}
 	}
-
-	return nil
+	once.Do(finish)
```

Comment:
The thing is that I want to wait here until goroutine with Done would finish and see if there was cancel to return "cancel" error. However, there's close in defer too to just shut down that goroutine on error. I might overlook some obvious solution, will appreciate help.
I see that logic is little flawed, probably waiting loop should be at one level up.